clang/test/CheckedCRewriter/b10_allsafepointerstruct.c

```diff
@@ -51,7 +51,7 @@ struct p *foo() {
   x->y = &ex2;
   y->y = &ex1;
   struct p *z = (struct p *) sus(x, y);
-	//CHECK: _Ptr<struct p> z =  (struct p *) sus(x, y);
+	//CHECK: _Ptr<struct p> z =  (_Ptr<struct p> ) sus(x, y);
```

Comment:
It occurs to me: In the case of `malloc` I think the cast is useful (right?), but in the above case it would make sense to delete the cast altogether. That's because the return type of `sus` and the expected type of `z` match up. Would it ever be the case that they would *not* match up but in a way that requires a C-style cast? 

You could imagine converting from an array to a pointer, and using the cast in that case, but I'm pretty sure Checked C just allows that change without a cast.

Reply:
That's an interesting point, I'm not sure when a c-style cast is required. I feel like an expression would either type check fine on its own or require a checked-c bounds cast to work. I don't think the cast on `malloc` is even useful. The cast on `malloc` is used to obtain the correct type argument instantiation, but once that's done, the type of `malloc` is known without any casting required. 

If we wanted to, we could probably drop casts during conversion, but I think we should prefer keeping them if they don't cause any problems.

Reply:
I agree that it's probably not needed on `malloc` either. Leaving them sounds fine.